tests/apollo/util/bft.py

```diff
@@ -153,7 +153,8 @@ async def background_sender(arg1, arg2, *, task_status=trio.TASK_STATUS_IGNORED)
 
 def with_bft_network(start_replica_cmd, selected_configs=None, num_clients=None, num_ro_replicas=0,
                      rotate_keys=False, bft_configs=None, with_cre=False, publish_master_keys=False,
-                     retries=1):
+                     num_repeats=int(os.getenv("NUM_REPEATS", 1)),
+                     break_on_first_failure=os.getenv('BREAK_ON_FAILURE') == 'TRUE', break_on_first_success=False):
```

Comment:
This will raise a keyerror if the environment variable is not set, which will be the case for most people.
You can use the second parameter of getenv for a default value.

Reply:
That is incorrect.
Ir returns 'None' if nothing is set as default, and 'None' does not equal 'True'.

Reply:
Since you pass it to a function whose parameter type is a bool, i think that this:
bool(os.getenv('BREAK_ON_FAILURE', False)) makes more sense

Reply:
bool(os.getenv('BREAK_ON_FAILURE', False)) Makes more sense since you pass to a function which expects a bool

Reply:
We are setting BREAK_ON_FAILURE=TRUE, but for readbility reasons one might choose to set  BREAK_ON_FAILURE=FALSE.
Sothis is not a good idea...